code/__HELPERS/qdel.dm

```diff
@@ -5,7 +5,11 @@
 #define QDEL_LIST_IN(L, time) addtimer(CALLBACK(GLOBAL_PROC, .proc/______qdel_list_wrapper, L), time, TIMER_STOPPABLE)
 #define QDEL_LIST_ASSOC(L) if(L) { for(var/I in L) { qdel(L[I]); qdel(I); } L.Cut(); }
 #define QDEL_LIST_ASSOC_VAL(L) if(L) { for(var/I in L) qdel(L[I]); L.Cut(); }
-
+#define QDEL_NULL_IN(obj, var, time) addtimer(CALLBACK(GLOBAL_PROC, .proc/______qdel_null_wrapper, obj, var), time, TIMER_STOPPABLE)
 
 /proc/______qdel_list_wrapper(list/L) //the underscores are to encourage people not to use this directly.
-	QDEL_LIST(L)
\ No newline at end of file
+	QDEL_LIST(L)
+
+/proc/______qdel_null_wrapper(datum/D, var_name)
+	qdel(D.vars[var_name])

```

Comment:
Given this is a timed action, check for `if(QDELETED(D))`?

Reply:
no harm in qdeling something already qdelled

Reply:
But what about accessing its vars, through `D.vars[var_name] = null`?
If `D` is no more, it will runtime.

Reply:
if D is qdelled then the timer will have been deleted also

Reply:
Alright.